code/modules/clothing/head/helmet.dm

```diff
@@ -115,3 +139,147 @@ obj/item/clothing/head/helmet/bluetaghelm
 	armor = list(melee = 30, bullet = 10, laser = 20,energy = 10, bomb = 20, bio = 0, rad = 0)
 	// Offer about the same protection as a hardhat.
 	flags_inv = HIDEEARS|HIDEEYES
+	canAttachCam =  0
+
+//LightToggle
+
+/obj/item/clothing/head/helmet/update_icon()
+
+	var/state = "[initial(icon_state)]"
+	if(helmetCam)
+		state += "-cam" //"helmet-cam"
+	if(F)
+		if(F.on)
+			state += "-flight-on" //"helmet-flight-on" // "helmet-cam-flight-on"
+		else
+			state += "-flight" //etc.
+
+	icon_state = state
+
+	usr.update_inv_head(0)
+	return
+
+/obj/item/clothing/head/helmet/ui_action_click()
+	toggle_helmlight()
+
+/obj/item/clothing/head/helmet/attackby(var/obj/item/A as obj, mob/user as mob, params)
+	if(istype(A, /obj/item/device/flashlight/seclite))
+		var/obj/item/device/flashlight/seclite/S = A
+		if(can_flashlight)
+			if(!F)
+				user.drop_item()
+				user << "<span class='notice'>You click [S] into place on [src].</span>"
+				if(S.on)
+					SetLuminosity(0)
+				F = S
+				A.loc = src
+				update_icon()
+				update_helmlight(user)
+				verbs += /obj/item/clothing/head/helmet/proc/toggle_helmlight
+		return
+
+	if(istype(A, /obj/item/weapon/screwdriver))
+		if(F)
+			for(var/obj/item/device/flashlight/seclite/S in src)
+				user << "<span class='notice'>You unscrew the seclite from [src].</span>"
+				F = null
+				S.loc = get_turf(user)
+				update_helmlight(user)
+				S.update_brightness(user)
+				update_icon()
+				usr.update_inv_head(0)
+				verbs -= /obj/item/clothing/head/helmet/proc/toggle_helmlight
+			return
+
+
+	if(istype(A, /obj/item/weapon/camera_assembly))
+		if(!canAttachCam)
+			user << "<span class='warning'>You can't attach [A] to [src]!</span>"
+			return
+		if(helmetCam)
+			user << "<span class='notice'>[src] already has a mounted camera.</span>"
+			return
+		user.drop_item()
+		helmetCam = new /obj/machinery/camera(src)

```

Comment:
Having it be a feature the AI can use would be beneficial for both the officer and the AI itself, but it would seem this is some rather iffy code. In practice, following a borg on its travels incurs a small delay akin to using the 'follow' command as a ghost.

Having it work for the AI is only worth it if it does not cause significant lag. Could a check be added to the camera's process() which has it update only if the loc var has changed and a certain amount of time has passed since the last update? (Maybe 2-3 seconds)

If that is not feasible, then keep it to Sec cameras only.